scipy/optimize/tests/test_linprog.py

```diff
@@ -1348,51 +1339,77 @@ def test_network_flow(self):
 
 
 class TestLinprogSimplexDefault(LinprogSimplexTests):
-    options = {}
 
-    def test_bug_7237(self):
-        # we want to check for the error
+    def setup_method(self):
+        self.options = {}
+
+    def test_bug_5400(self):
+        with pytest.raises(ValueError):
+            super(TestLinprogSimplexDefault, self).test_bug_5400()
+
+    def test_bug_7237_low_tol(self):
+        # Fails if the tolerance is too strict. Here we test that
+        # even if the solutuion is wrong, the appropriate error is raised.
+        self.options.update({'tol': 1e-12})
         with pytest.raises(ValueError):
             super(TestLinprogSimplexDefault, self).test_bug_7237()
 
-    def test_bug_8174(self):
-        # we want to check for the warning about the pivot
+    def test_bug_8174_low_tol(self):
+        # Fails if the tolerance is too strict. Here we test that
+        # even if the solutuion is wrong, the appropriate warning is issued.
+        self.options.update({'tol': 1e-12})
         with pytest.warns(OptimizeWarning):
             super(TestLinprogSimplexDefault, self).test_bug_8174()
 
 
 class TestLinprogSimplexBland(LinprogSimplexTests):
-    options = {'bland': True}
 
-    # we want to check for the error
+    def setup_method(self):
+        self.options = {'bland': True}
+
     def test_bug_5400(self):
         with pytest.raises(ValueError):
             super(TestLinprogSimplexBland, self).test_bug_5400()
 
-    def test_bug_8174(self):
-        with pytest.warns(OptimizeWarning):
-            pytest.skip("This fails, but it shouldn't.")
+    def test_bug_8174_low_tol(self):
+        # Fails if the tolerance is too strict. Here we test that
+        # even if the solutuion is wrong, the appropriate error is raised.
+        self.options.update({'tol': 1e-12})
+        with pytest.raises(AssertionError):
+            with pytest.warns(OptimizeWarning):
+                super(TestLinprogSimplexBland, self).test_bug_8174()
 
 
 class TestLinprogSimplexNoPresolve(LinprogSimplexTests):
-    options = {'presolve': False}
 
-    def test_bug_6139(self):
+    def setup_method(self):
+        self.options = {'presolve': False}
+
+    @pytest.mark.xfail(reason='Fails with warning on 32-bit linux')
```

Comment:
If the failure is only on 32-bit Linux you could make this mark conditional on that scenario.

Reply:
Good point. I'm not sure if xfail supports skipping based on certain conditions, but there is [``pytest.mark.skipif``](https://docs.pytest.org/en/latest/reference.html#pytest-mark-skipif-ref). Would skipping the test be acceptable?

Reply:
xfail does support a condition: https://docs.pytest.org/en/latest/reference.html#pytest-mark-xfail-ref

There are almost certainly examples of 32-bit selective skipping around in NumPy / SciPy test suites. Presumably a short check like here: https://stackoverflow.com/a/1405971/2942522

Reply:
Sorry for the noise. I know there is a way of doing it. 

I wanted to check if there is a neater (or standard) way of doing it. Using ``sys.maxsize > 2 ** 32`` seems like a roundabout way of doing it. Even when there are examples in NumPy/SciPy they can be inconsistent or not the best way of doing it. I'd prefer to get it  right the first time.

Reply:
`scipy/special/tests/test_trig.py` has

```python
@pytest.mark.xfail('win32' in sys.platform
                   and np.intp(0).itemsize < 8
                   and sys.version_info < (3, 5),
                   reason="fails on 32-bit Windows with old MSVC")
```
and `scipy/spatial/tests/test_qhull.py`  has
`_is_32bit_platform = np.intp(0).itemsize < 8`

So that seems like a decent start